apps/fortuna/src/command/run.rs

```diff
@@ -218,11 +240,158 @@ pub async fn run(opts: &RunOptions) -> Result<()> {
 
     Ok(())
 }
+
+/// tracks the balance of the given address for each chain in the given config periodically
+pub async fn track_balance(
+    config: Config,
+    address: Address,
+    metrics_registry: Arc<metrics::Metrics>,
+) {
+    loop {
+        for (chain_id, chain_config) in &config.chains {
+            let provider = match Provider::<Http>::try_from(&chain_config.geth_rpc_addr) {
+                Ok(r) => r,
+                Err(_e) => continue,
+            };
+
+            let balance = match provider.get_balance(address, None).await {
+                // This conversion to u128 is fine as the total balance will never cross the limits
+                // of u128 practically.
+                Ok(r) => r.as_u128(),
+                Err(_e) => continue,
+            };
+            // The f64 conversion is made to be able to serve metrics with the constraints of Prometheus.
+            // The balance is in wei, so we need to divide by 1e18 to convert it to eth.
+            let balance = balance as f64 / 1e18;
+
+            metrics_registry
+                .balance
+                .get_or_create(&AccountLabel {
+                    chain_id: chain_id.clone(),
+                    address:  address.to_string(),
+                })
+                // comment on why is this ok
+                .set(balance);
+        }
+
+        time::sleep(TRACK_DURATION).await;
+    }
+}
+
+/// tracks the collected fees of the given address for each chain in the given config periodically
+pub async fn track_collected_fee(
+    config: Config,
+    provider_address: Address,
+    metrics_registry: Arc<metrics::Metrics>,
+) {
+    loop {
+        for (chain_id, chain_config) in &config.chains {
+            let contract = match PythContract::from_config(chain_config) {
+                Ok(r) => r,
+                Err(_e) => continue,
+            };
+
+            let provider_info = match contract.get_provider_info(provider_address).call().await {
+                Ok(info) => info,
+                Err(_e) => {
+                    time::sleep(Duration::from_secs(5)).await;
+                    continue;
+                }
+            };
+
+            // The f64 conversion is made to be able to serve metrics with the constraints of Prometheus.
+            // The fee is in wei, so we need to divide by 1e18 to convert it to eth.
+            let collected_fee = provider_info.accrued_fees_in_wei as f64 / 1e18;
+
+            metrics_registry
+                .collected_fee
+                .get_or_create(&AccountLabel {
+                    chain_id: chain_id.clone(),
+                    address:  provider_address.to_string(),
+                })
+                .set(collected_fee);
+        }
+
+        time::sleep(TRACK_DURATION).await;
+    }
+}
+
+/// tracks the current sequence number and end sequence number of the given provider address for
+/// each chain in the given config periodically
+pub async fn track_hashchain(
```

Comment:
This function can be merged with `track_collected_fee` since they are basically making the same RPC call